src/cache.c

```diff
@@ -1128,7 +1128,7 @@ cache_artwork_add_impl(void *arg, int *retval)
 static enum command_state
 cache_artwork_get_impl(void *arg, int *retval)
 {
-#define Q_TMPL "SELECT a.format, a.data FROM artwork a WHERE a.type = %d AND a.persistentid = %" PRIi64 " AND a.max_w = %d AND a.max_h = %d;"
+#define Q_TMPL "SELECT a.format, a.data FROM artwork a WHERE a.type = %d AND a.persistentid = %" PRIi64 " AND a.max_w = %d AND a.max_h = %d AND LENGTH(data) IS NOT NULL;"
```

Comment:
When a client requests for artwork for a new album and this is the first time asking ever, a row is added to the `artwork` table but `data` blob column is empty.

Without this new clause the code incorrect states that we found a cached artwork even though the `data` blob returned is 0 length.